src/main/java/org/springframework/integration/aws/inbound/kinesis/KclMessageDrivenChannelAdapter.java

```diff
@@ -0,0 +1,408 @@
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchAsyncClient;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
+import software.amazon.awssdk.services.kinesis.KinesisAsyncClient;
+import software.amazon.kinesis.common.ConfigsBuilder;
+import software.amazon.kinesis.common.InitialPositionInStream;
+import software.amazon.kinesis.common.InitialPositionInStreamExtended;
+import software.amazon.kinesis.coordinator.Scheduler;
+import software.amazon.kinesis.exceptions.InvalidStateException;
+import software.amazon.kinesis.exceptions.ShutdownException;
+import software.amazon.kinesis.exceptions.ThrottlingException;
+import software.amazon.kinesis.lifecycle.events.InitializationInput;
+import software.amazon.kinesis.lifecycle.events.LeaseLostInput;
+import software.amazon.kinesis.lifecycle.events.ProcessRecordsInput;
+import software.amazon.kinesis.lifecycle.events.ShardEndedInput;
+import software.amazon.kinesis.lifecycle.events.ShutdownRequestedInput;
+import software.amazon.kinesis.processor.RecordProcessorCheckpointer;
+import software.amazon.kinesis.processor.ShardRecordProcessor;
+import software.amazon.kinesis.processor.ShardRecordProcessorFactory;
+import software.amazon.kinesis.retrieval.KinesisClientRecord;
+
+/**
+ * The {@link MessageProducerSupport} implementation for receiving data from Amazon
+ * Kinesis stream(s) using AWS KCL.
+ *
+ * @author Hervé Fortin
+ *
+ * @since 2.1.0
+ */
+@ManagedResource
+@IntegrationManagedResource
+public class KclMessageDrivenChannelAdapter extends MessageProducerSupport implements DisposableBean {
+
+	private final String stream;
+
+	private String consumerGroup = "SpringIntegration";
+
+	private InboundMessageMapper<byte[]> embeddedHeadersMapper;
+
+	private Scheduler scheduler;
+
+	private Executor executor;
+
+	private KinesisAsyncClient kinesisClient;
+
+	private CloudWatchAsyncClient cloudWatchClient;
+
+	private DynamoDbAsyncClient dynamoDBClient;
+
+	private InitialPositionInStreamExtended streamInitialSequence =
+			InitialPositionInStreamExtended.newInitialPosition(InitialPositionInStream.LATEST);
+
+	private int idleBetweenPolls;
+
+	private int consumerBackoff;
+
+	private long checkpointsInterval = 60_000L;
+
+	public KclMessageDrivenChannelAdapter(String streams, Executor executor) {
+		this(streams, executor, KinesisAsyncClient.builder().build(),
+				CloudWatchAsyncClient.builder().build(), DynamoDbAsyncClient.builder().build());
+	}
+
+	public KclMessageDrivenChannelAdapter(String streams, Executor executor, Region region) {
+		this(streams, executor, KinesisAsyncClient.builder().region(region).build(),
+				CloudWatchAsyncClient.builder().region(region).build(), DynamoDbAsyncClient.builder().region(region).build());
+	}
+
+	public KclMessageDrivenChannelAdapter(String streams, Executor executor,
```

Comment:
Looks like we deal here only with a single `srtream`, so, this param has to be singular anyway.